internals/overlord/overlord.go

```diff
@@ -89,7 +88,7 @@ type Overlord struct {
 	ensureLock  sync.Mutex
 	ensureTimer *time.Timer
 	ensureNext  time.Time
-	ensureRun   int32
+	ensureRun   chan struct{}
```

Comment:
I am extending an existing feature here to also allow for waiting, not only checking if ensure ran already.